currencies/currency_converter_test.go

```diff
@@ -0,0 +1,557 @@
+package currencies_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prebid/prebid-server/currencies"
+)
+
+func TestFetch_Success(t *testing.T) {
+
+	// Setup:
+	requestURL := "currency.fake.com/latest.json"
+	responseBody := `{
+		"dataAsOf":"2018-09-12",
+		"conversions":{
+			"USD":{
+				"GBP":0.77208
+			},
+			"GBP":{
+				"USD":1.2952
+			}
+		}
+	}`
+
+	mockedHttpClient := &HttpClientMock{
+		responseStatusCode: http.StatusOK,
+		responseBody:       responseBody,
+		err:                nil,
+	}
+
+	rateConverter := currencies.NewRateConverter(
+		mockedHttpClient,
+		requestURL,
+		time.Duration(1)*time.Minute,
+		time.Duration(5)*time.Minute,
+	)
+
+	expectedDataAsOf := time.Date(2018, time.September, 12, 0, 0, 0, 0, time.UTC)
+	expectedRates := currencies.Rates{
+		DataAsOf:    &expectedDataAsOf,
+		DataAsOfRaw: "2018-09-12",
+		Conversions: map[string]map[string]float32{
+			"USD": {
+				"GBP": 0.77208,
+			},
+			"GBP": {
+				"USD": 1.2952,
+			},
+		},
+	}
+
+	// Execute:
+	beforeExecution := time.Now()
+	rates, err := rateConverter.Update()
+
+	// Verify:
+	if !mockedHttpClient.VerifyUrlCalled(requestURL) {
+		t.Errorf("sync URL `%s` wasn't called", requestURL)
+		return
+	}
+	if !mockedHttpClient.VerifyHasNoErrors() {
+		t.Error("expects no error while fecthing remote source")
+		return
+	}
+	if err != nil {
```

Comment:
Check out Testify: https://godoc.org/github.com/stretchr/testify/assert

It's already a project dependency, and makes tests so much cleaner.

Reply:
yes, thanks :)